packages/jest-reporters/src/get_result_header.ts

```diff
@@ -46,7 +46,7 @@ export default (
     : null;
 
   const testDetail = [];
-  if (runTime !== null && runTime > 5) {
+  if (runTime !== null && runTime > globalConfig.longTestThreshold) {
```

Comment:
As OP from the mentioned issue said, such config option makes sense for specific scenarios (like running e2e tests), or in Jest nomenclature "projects". Hence, putting it into `globalConfig` is not the best place. 

I'd go with either putting this inside `projectConfig` or making it an option of a reporter as @SimenB suggested. I'm not 100% convinced where the responsibility lies, but for now I'd expect Jest to determine whether the test was slow and pass that information to the reporter to render, so I'm leaning towards `projectConfig` solution.